### PR TITLE
feat(component_state_monitor): add component state monitor

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Add component state monitor package. This package is an extract of ad_service_state_monitor's topic monitoring functionality.

## Related links

https://github.com/tier4/tier4_autoware_msgs/pull/61

## Tests performed

1. Check launch ready state topics are all true when Autoware is launched.
    `ros2 topic echo /system/component_state_monitor/component/launch/<component>`
2. Check autonomous ready state of localization topic is true after initialpose.
    `ros2 topic echo /system/component_state_monitor/component/autonomous/localization`
3. Check autonomous ready state of planning/control topics are true after route setting.
    `ros2 topic echo /system/component_state_monitor/component/autonomous/planning`
    `ros2 topic echo /system/component_state_monitor/component/autonomous/control`

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: system/component_state_monitor/config/topics.yaml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    node_name_suffix: pose_twist_fusion_filter_pose
+    topic: /localization/pose_twist_fusion_filter/pose
+    topic_type: geometry_msgs/msg/PoseStamped
+    best_effort: false
+    transient_local: false
+    warn_rate: 5.0
+    error_rate: 1.0
+    timeout: 1.0
+
+- module: sensing
```
Comment (yukkysaito) [inline system/component_state_monitor/config/topics.yaml:53]: obstacle segmentation is part of perception.

File: system/component_state_monitor/config/topics.yaml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  args:
+    node_name_suffix: object_recognition_objects
+    topic: /perception/object_recognition/objects
+    topic_type: autoware_auto_perception_msgs/msg/PredictedObjects
+    best_effort: false
+    transient_local: false
+    warn_rate: 5.0
+    error_rate: 1.0
+    timeout: 1.0
+
```
Comment (yukkysaito) [inline system/component_state_monitor/config/topics.yaml:78]: Can traffic light recognition results be added?

File: system/component_state_monitor/config/topics.yaml (hunk 1 above)
Comment (isamu-takagi) [reply system/component_state_monitor/config/topics.yaml:53]: The [original setting](https://github.com/autowarefoundation/autoware.universe/blob/main/system/ad_service_state_monitor/config/ad_service_state_monitor.param.yaml#L55-L56) is sensing. Can I change it?

File: system/component_state_monitor/config/topics.yaml (hunk 2 above)
Comment (isamu-takagi) [reply system/component_state_monitor/config/topics.yaml:78]: What is the topic name and expected rate?

File: system/component_state_monitor/config/topics.yaml (hunk 1 above)
Comment (yukkysaito) [reply system/component_state_monitor/config/topics.yaml:53]: Yes. It is moved from sensing to perception during the core/universe transplant.

**Review by kenji-miyake — APPROVED** [review]
Thanks! If it works, it looks good to me for Autoware Universe.

For Autoware Core, @xmfcx is reconsidering a monitoring architecture, so we should discuss it later.
https://github.com/autowarefoundation/autoware.core/pull/57

File: system/component_state_monitor/config/topics.yaml (hunk 1 above)
Comment (isamu-takagi) [reply system/component_state_monitor/config/topics.yaml]: Done https://github.com/autowarefoundation/autoware.universe/pull/2120/commits/fa5f10655d80a2d08e994d435bca9809217de2e5.

File: system/component_state_monitor/config/topics.yaml (hunk 2 above)
Comment (isamu-takagi) [reply system/component_state_monitor/config/topics.yaml:78]: In planning simulation, the topic was not published periodically, so I will merge it once. Please change the config if necessary.

**Pull request merged**